### PR TITLE
Nitya - Homework3

### PR DESCRIPTION
**Pull request by benkahle**
Graded by @benkahle and @EvanDorsky

### CODE REVIEW

File: classes/class1/homework/learnyounode/node_modules/bl/.jshintrc [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
```
Comment (benkahle) [inline classes/class1/homework/learnyounode/node_modules/bl/.jshintrc]: Do not push `node_modules` directories to git!
Each user should install them independently to ensure getting the correct version and platform for their dev environment. Also, it clouds your commit history (like this pull request with more than 2,300 files).

File: classes/class3/catAppHw/app.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+//jump to any directory, static checks path of request
+//inside public. so i can get test.txt navigate to localhost:3000/javascripts/text.txt
+//-----------------------------------------------
+
+app.get('/cats/new', index.catsNew);
+app.get('/cats', index.cats);
+app.get('/cats/delete/old', index.deleteOldCat);
+app.get('/cats/bycolor/:color', index.catsByColor);
+
+app.get('/olin', function(req,res,next){
```
Comment (benkahle) [inline classes/class3/catAppHw/app.js:39]: You should remove some of the class code that is not related to this app. In general remove debugging and unused code before deploying and submitting code.

File: classes/class3/catAppHw/routes/index.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
+var Cat = require('./../models/catModel.js');
+
+
+var cats = function(req, res){
+
+	Cat.find(function(err, cats){
+		if (err){
+			console.log("can't find cats", err);
+		}
+		var catsArr = cats;
```
Comment (benkahle) [inline classes/class3/catAppHw/routes/index.js:10]: You can just use the `cats` variable directly.

File: classes/class3/catAppHw/routes/index.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+		res.render("cats", {catslist:catsArr});
```
Comment (benkahle) [inline classes/class3/catAppHw/routes/index.js:12]: You may want to render something different in the case of an error.

File: classes/class3/catAppHw/routes/index.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+// 		if 
+// 		Cat.findOneAndRemove({'name': toDel.name}, function(err, data){
+// 			if(err){
+// 				console.log("couldn't remove cats", err);
+// 			}
+// 			res.render("cats", {catslist:[]});
+// 		});
+
+// 	});
+// }
```
Comment (EvanDorsky) [inline classes/class3/catAppHw/routes/index.js:67]: Generally when you use version control (like Git), it's better to delete large blocks of code than to comment them out.

File: classes/class3/catAppHw/routes/index.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+		age:randAge, 
+		colors: catCol});
+
+	nCat.save(function(err){
+		if(err){
+			console.log("problemo saving nCat", err);
+		}
+	});
+
+	res.render("cats", {catslist:[nCat]});
```
Comment (EvanDorsky) [inline classes/class3/catAppHw/routes/index.js:46]: Good reuse of templates.

File: classes/class3/catAppHw/routes/index.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+	var randCol2 = colorsArr[Math.floor(colorsArr.length*Math.random())];
+	if (randCol1 !==randCol2){
+		var catCol = [randCol1, randCol2];
+	}
+	else{
+		var catCol = [randCol1];
+	}
+	var randAge = Math.floor(20*Math.random())+1;//rand from 1-20
+
+	var nCat = new Cat({name:randName,
```
Comment (benkahle) [inline classes/class3/catAppHw/routes/index.js:36]: Generally, start parameters on a new line to align them nicely.

File: classes/class3/catAppHw/routes/index.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+	}
+	var randAge = Math.floor(20*Math.random())+1;//rand from 1-20
+
+	var nCat = new Cat({name:randName, 
+		age:randAge, 
+		colors: catCol});
+
+	nCat.save(function(err){
+		if(err){
+			console.log("problemo saving nCat", err);
```
Comment (benkahle) [inline classes/class3/catAppHw/routes/index.js:42]: You should tell the user about the error.

File: classes/class3/catAppHw/routes/index.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+	Cat.findOneAndRemove({}, {sort: {age:-1}}, function(err, cat){
+		if(err){
+			console.log("oh no");
+		}
+		res.render("cats", {catslist:[]});
+	});
+}
+
+
+module.exports.deleteOldCat = deleteOldCat
```
Comment (benkahle) [inline classes/class3/catAppHw/routes/index.js:79]: You should group your exports into an object and export them all together.
